src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,30 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserException("Incorrect value");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserException("A user with this login is already registered");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new InvalidUserException("Login must contain at least 6 characters");
+        }
+        if (user.getPassword().length() < 6) {
+            throw new InvalidUserException("Password must contain at least 6 characters");
+        }
+        if (user.getAge() < 18) {
+            throw new InvalidUserException("You are not have 18 years old");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidUserException("Login cannot be empty");
```

Comment:
The null check for 'login' should come before any other checks that use the 'login' field to prevent NullPointerException.